### PR TITLE
Updates the class-based resource in base-branch 'test-problem-class-based-resource-issue-#673'

### LINKED ISSUE #673: xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership: Logic error in tests
**Details of the scenario you tried and the problem that is occurring:**
When trying to opt-in for common tests, the tests for xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership started to fail..

See errors:
https://ci.appveyor.com/project/johlju/xsqlserver/build/6.0.700.0#L1717

```plaintext
System.InvalidOperationException: Neither PowerShell module SqlServer or SQLPS was found. Unable to run SQL Server cmdlets.
        at <ScriptBlock>, C:\projects\xsqlserver\DscResources\CommonResourceHelper.psm1: line 88
        88:     throw $errorRecordToThrow
```

I think it has to do with this row that loads Import-SQLPSModule in the construct on initialization.
```powershell
    # Load the SQLPS module when the class is initialized
    xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership()
    {
        Import-SQLPSModule
    }
```

That means that this code in the tests fail, because the mocks are only active when the It-block is run, not when BeforeEach is run. 

```powershell
            BeforeEach {
                $databaseMembershipClass = [xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership]::New()
                $databaseMembershipClass.DatabaseName = $mockDatabaseNameParameter.Clone()
                $databaseMembershipClass.SqlServer = 'Server1'
                $databaseMembershipClass.SQLInstanceName = 'MSSQLSERVER'
                $databaseMembershipClass.AvailabilityGroupName = 'AvailabilityGroup1'
            }
```

The downside of having that in the construct is that each call to initialize a new object will call Import-SQLPSModule regardless if it is needed or not.
```powershell
$currentConfiguration = New-Object xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership 
```

But the above seems wrong to, because then the tests would work at all. But they do, unless I opt-in for the common tests. What that does is activate markdown checking and example compiling. And I think it is the last that changes this behavior in some way. 

BUT, I can repeat this error in the command line. When running the test in a new PowerShell window, the tests works, but if I run it a second time, the test fail with the same error as above. If I change the code, like just a white space, then the test runs fine on a third try, but when I then run a forth time it fails again, and so on. Which make me conclude that something fishy is going on with Pester and class-based resources.

**The DSC configuration that is using the resource (as detailed as possible):**
n/a

**Version of the Operating System, SQL Server and PowerShell the DSC Target Node is running:**
n/a

**What module (SqlServer or SQLPS) and which version of the module the DSC Target Node is running:**
n/a

**Version of the DSC module you're using, or 'dev' if you're using current dev branch:**
Dev

### PR DESCRIPTION
**Pull request by randomnote1**
**Pull Request (PR) description**
Fix module manifest for class-based resources and logic errors in tests.

**This Pull Request (PR) fixes the following issues:**
Fixes #673 

**Task list:**
- [ ] Change details added to Unreleased section of CHANGELOG.md?
- [ ] Added/updated documentation, comment-based help and descriptions in .schema.mof files where appropriate?
- [ ] Examples appropriately updated?
- [X] New/changed code adheres to [Style Guidelines](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md)?
- [ ] [Unit and (optional) Integration tests](https://github.com/PowerShell/DscResources/blob/master/TestsGuidelines.md) created/updated where possible?

---
This change is [Reviewable](https://reviewable.io/reviews/powershell/xsqlserver/678)

### CODE REVIEW

**Comment by johlju** [comment]
@randomnote1 The .psd1 file seems to be binary.

**Comment by randomnote1** [comment]
For the initial review of this, I left the class-based resource in the DSCResources directory. I'm curious to see if this alone resolves the issue of being able to run the tests multiple times.

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/678#-:-KohOMV_QunY1VcBWxxl:b-95ewb1)*

**Comment by randomnote1** [comment]
Which .psd1? They both look fine to me...

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/678#-:-KohOiphURXdwKzIg3Rf:b-t4uljg)*

**Comment by johlju** [comment]
image

I think it is the file encoding that is wrong. 😄

**Comment by randomnote1** [comment]
oh, I see. What module is that ConvertTo-Utf8 cmdlet in? I can never find it :(

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/678#-:-KohSDGu8TpHCU7nRZXZ:b-v86ual)*

**Comment by johlju** [comment]
It's under DscResource.Tests\MetaFixers.psm1

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/678#-:-KohSrRicmE15U0GkVvW:b-z0sod0)*

**Comment by randomnote1** [comment]
Got it, thanks

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/678#-:-KohU2fwcqpWXSARZtyz:b-vo0hen)*

**Comment by johlju** [comment]
Didn't seem to help with running the tests twice. :/ Does it work better with the DSCClassResource folder?

**Comment by randomnote1** [comment]
Nope, that didn't help either. I'll merge my test with the DSCClassResource folder.

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/678#-:-Kohdxq3NsTaW8HXWQkn:b-yfkoj1)*

**Comment by randomnote1** [comment]
One thing I did for quick testing was copy the `CommonResourceHelper.psm1` file to the `DSCClassResources` directory. Just an FYI for later.

---

Review status: 0 of 9 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/678#-:-KohtH27Sx9qkHl-ScHf:b-zihppa)*

**Comment by johlju** [comment]
Yes saw that. :) We do clean up afterwards :)

It's getting late here so comment if you want me to do anything/test anything tomorrow.

**Comment by johlju** [comment]
This still can't be tested twice. :/ What do you say about splitting the logic to another file and test that and see if that works better? 

You can remove MSFT_ prefix from all the files for this class based resource.

Also rebase this branch so we get the latest changes (less changes in the changelog)

---

Review status: 0 of 9 files reviewed at latest revision, 6 unresolved discussions.

---

*[appveyor.yml, line 6 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/678#-KokL4NsQHo3rjzgfja_:-KokL4NsQHo3rjzgfjaa:b-h7x21c) ([raw file](https://github.com/powershell/xsqlserver/blob/1c5bdba81a7d2485bfa2b3f545dacdc6257c388e/appveyor.yml#L6)):*
> ```YAML
> version: 6.0.{build}.0
> install:
>     - git clone https://github.com/johlju/DscResource.Tests
> ```

Katie approved this change. Hopefully we get it merged today.

---

*[xSQLServer.psd1, line 33 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/678#-KokJjC3RIkFoScDgv2k:-KokJjC3RIkFoScDgv2l:b-ysw96b) ([raw file](https://github.com/powershell/xsqlserver/blob/1c5bdba81a7d2485bfa2b3f545dacdc6257c388e/xSQLServer.psd1#L33)):*
> ```PowerShell
> 
> DSCResourcesToExport = @(
>   'xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership'
> ```

This makes no difference if it set to the resource name, regardless if NestedModule is pointing to the .psm1 or .psd1 file.
I'm curious what @kwirkykat meant by this is her comment [here](https://github.com/PowerShell/xSQLServer/issues/425#issuecomment-298436623), since it seems to be the most important. Which get's me wondering if something else is wrong too. :smile:
Because if you remove NestedModules it no longer detects the class-based resource, so that is the only property (so far) making this work.

---

*[xSQLServer.psd1, line 38 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/678#-KokLcdkxcOnXPnW5Js4:-KokLcdkxcOnXPnW5Js5:b1sankg) ([raw file](https://github.com/powershell/xsqlserver/blob/1c5bdba81a7d2485bfa2b3f545dacdc6257c388e/xSQLServer.psd1#L38)):*
> ```PowerShell
> # Modules to import as nested modules of the module specified in RootModule/ModuleToProcess
> 
> NestedModules = @(
> ```

Without this nothing works. This seems to be the most important one! This is needed for the resource to be detected. This also enables more class-based resources. 

---

*[DSCClassResources/CommonResourceHelper.psm1, line 1 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/678#-KokLAx7sfL-sHme3kmA:-KokLAx8mleK0p2Vy7TD:b25ojwo) ([raw file](https://github.com/powershell/xsqlserver/blob/1c5bdba81a7d2485bfa2b3f545dacdc6257c388e/DSCClassResources/CommonResourceHelper.psm1#L1)):*
> ```PowerShell
> <#
> ```

Just commenting here for a reminder that this file should be removed, and the resource should instead point to the CommonResourceHelper under DscResource. But this might be obsolete if we need to split the logic for the class-based resource.

---

*[DSCClassResources/MSFT_xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership/MSFT_xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership.psm1, line 83 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/678#-KokLvjvTJdQoDN5xMb6:-KokLvjvTJdQoDN5xMb7:b-84fu84) ([raw file](https://github.com/powershell/xsqlserver/blob/1c5bdba81a7d2485bfa2b3f545dacdc6257c388e/DSCClassResources/MSFT_xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership/MSFT_xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership.psm1#L83)):*
> ```PowerShell
>     [Void] Set()
>     {
>         Import-SQLPSModule
> ```

Doesn't this need to be in the Get or Test method?

---

*[DSCClassResources/MSFT_xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership/xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership.psd1, line 1 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/678#-KokI97jN4jfa7bmus0s:-KokI97jN4jfa7bmus0t:bppt7gv) ([raw file](https://github.com/powershell/xsqlserver/blob/1c5bdba81a7d2485bfa2b3f545dacdc6257c388e/DSCClassResources/MSFT_xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership/xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership.psd1#L1)):*
> ```PowerShell
> @{
> ```

This file seems only be needed if NestedModule is pointed to it in the module manifest xSQLServer.psd1 (instead of the .psm1).
And then, the only properties needed seem to be RootModule (or NestedModule) and ModuleVersion. The others make no difference when running Get-DscResource and compiling a configuration.

So since nested module is pointing to the .psm1 I guess we can remove this? Then we don't have to update the version number in yet another file.
What do you think?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/678#-:-KokIu_hY3rg0UMbhYFY:b396y57)*

**Comment by johlju** [comment]
@randomnote1 I split the Get()-method into another function Get-TargetResource, and now I can run the Get-test every time!
It looks like we need to do this! I would be more comfortable if you did this change, since you have better knowledge of the resource. Are you okay with that?

I could not push to this PR, so I pushed it to my fork in the branch 'split-get-method'.
https://github.com/johlju/xSQLServer/blob/split-get-method/DSCClassResources/MSFT_xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership/MSFT_xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership.psm1#L667

I had a problem with making this parameter mandatory since the tests for some reason was allowed to pass a null value to the Get() method even when the parameter was mandatory. It was not allowed to pass the $null value to the function Get-TargetResource when the parameter was mandatory. So I guess the tests should be changed. It was faster just setting this to be non-mandatory, I didn't want to make to big of a change.
```
        [Parameter()] #[Parameter(Mandatory = $true)]
        [System.String]
        $BackupPath,
```

**Comment by randomnote1** [comment]
I wonder if we'll see odd behaviour because get/set/test are defined both as methods on the object and functions.

I feel like if we go down the route to move the contents of the methods into functions we might as well convert it to a MOF based resource. I'd rather do that with proper unit tests than have a class based resource that is hacked together soley to pass the unit tests. Thoughts?

We also haven't answered the question as to why functions are only mocked the first time Pester is executed on a class. This is something I think needs to be investigated and fixed for the bigger picture here.

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/678#-:-KolO5a9lETDuHky99ia:bki2axd)*

**Comment by randomnote1** [comment]
Review status: 0 of 9 files reviewed at latest revision, 6 unresolved discussions.

---

*[xSQLServer.psd1, line 33 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/678#-KokJjC3RIkFoScDgv2k:-KolSI-nWdVQ4MijNHyK:bv63xv) ([raw file](https://github.com/powershell/xsqlserver/blob/1c5bdba81a7d2485bfa2b3f545dacdc6257c388e/xSQLServer.psd1#L33)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

This makes no difference if it set to the resource name, regardless if NestedModule is pointing to the .psm1 or .psd1 file.
I'm curious what @kwirkykat meant by this is her comment [here](https://github.com/PowerShell/xSQLServer/issues/425#issuecomment-298436623), since it seems to be the most important. Which get's me wondering if something else is wrong too. :smile:
Because if you remove NestedModules it no longer detects the class-based resource, so that is the only property (so far) making this work.
</blockquote></details>

Withouth the DSCResourcesToExport property, the resource does not appear when `Get-DscResource -Module xSQLServer` is executed.

---

*[DSCClassResources/MSFT_xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership/MSFT_xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership.psm1, line 83 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/678#-KokLvjvTJdQoDN5xMb6:-KolSh5nqogR4r0sEmGx:b-rj8ebn) ([raw file](https://github.com/powershell/xsqlserver/blob/1c5bdba81a7d2485bfa2b3f545dacdc6257c388e/DSCClassResources/MSFT_xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership/MSFT_xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership.psm1#L83)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Doesn't this need to be in the Get or Test method?
</blockquote></details>

It shouldn't. I'm not calling any cmdlets defined in Import-SQLPSModule in those methods.

---

*[DSCClassResources/MSFT_xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership/xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership.psd1, line 1 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/678#-KokI97jN4jfa7bmus0s:-KolU57LvYaTOuQid8PF:b-r6dkw1) ([raw file](https://github.com/powershell/xsqlserver/blob/1c5bdba81a7d2485bfa2b3f545dacdc6257c388e/DSCClassResources/MSFT_xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership/xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership.psd1#L1)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

This file seems only be needed if NestedModule is pointed to it in the module manifest xSQLServer.psd1 (instead of the .psm1).
And then, the only properties needed seem to be RootModule (or NestedModule) and ModuleVersion. The others make no difference when running Get-DscResource and compiling a configuration.

So since nested module is pointing to the .psm1 I guess we can remove this? Then we don't have to update the version number in yet another file.
What do you think?
</blockquote></details>

The psd1 appears to be required in order for Get-DscResource to function.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/678)*

**Comment by johlju** [comment]
I think the reason the Pester test only works the first time, is that on the second run the class is already loaded, and for some reason Pester fails. When I add the common tests for examples and markdown files from DscResource.Test, it copies the resource to the $PSMoudlePath and it gets loaded (I think). And because of the class is loaded the unit test fails. See PR #681 where the tests fail with the same error as when we run them the second time on command line (this PR must work with passing tests).

I thought is that Pester not working correctly with classes is a bigger issue than just this resource. We could file issues in Pester repo referencing a working branch with the current state. But we need a workaround in the meantime so we can release this resource with xSQLServer (to me that means working tests for PR #681).

I'm all for converting this resource to mof and getting proper tests! 

The only reason I see that we should keep this as a class-based resource and split the logic into other functions (so we could test the code) is that we could easily revert back the logic to the Get-/Set-/Test-methods when the Pester testing has been solved. But on the other hand, if we keep the current state in another working branch, we have both the class-based resource and unit tests for it to just PR in replacing a mof resource if we convert it.

**Comment by johlju** [comment]
Review status: 0 of 9 files reviewed at latest revision, 5 unresolved discussions.

---

*[xSQLServer.psd1, line 33 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/678#-KokJjC3RIkFoScDgv2k:-KolVSn3bO4jabDq7uQf:b-9obmgz) ([raw file](https://github.com/powershell/xsqlserver/blob/1c5bdba81a7d2485bfa2b3f545dacdc6257c388e/xSQLServer.psd1#L33)):*
<details><summary><i>Previously, randomnote1 (Dan Reist) wrote…</i></summary><blockquote>

Withouth the DSCResourcesToExport property, the resource does not appear when `Get-DscResource -Module xSQLServer` is executed.
</blockquote></details>

Hmm... Worked without it this morning for me. But if you see different result then lets keep it in. 

---

*[DSCClassResources/MSFT_xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership/xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership.psd1, line 1 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/678#-KokI97jN4jfa7bmus0s:-KolVpSI5K7ENZekhCwz:bm53b2r) ([raw file](https://github.com/powershell/xsqlserver/blob/1c5bdba81a7d2485bfa2b3f545dacdc6257c388e/DSCClassResources/MSFT_xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership/xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership.psd1#L1)):*
<details><summary><i>Previously, randomnote1 (Dan Reist) wrote…</i></summary><blockquote>

The psd1 appears to be required in order for Get-DscResource to function.
</blockquote></details>

Hmm... Worked without it this morning for me. But if you see different result then lets keep it in. But remove any properties not needed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/678)*

**Comment by johlju** [comment]
Review status: 0 of 9 files reviewed at latest revision, 5 unresolved discussions.

---

*[DSCClassResources/MSFT_xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership/xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership.psd1, line 1 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/678#-KokI97jN4jfa7bmus0s:-KolWDLt3iCs-2Y0hcYk:b-bhgpks) ([raw file](https://github.com/powershell/xsqlserver/blob/1c5bdba81a7d2485bfa2b3f545dacdc6257c388e/DSCClassResources/MSFT_xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership/xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership.psd1#L1)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Hmm... Worked without it this morning for me. But if you see different result then lets keep it in. But remove any properties not needed.
</blockquote></details>

Note that I changed to .psm1 for NestedModules instead of .psd1 when I removed this file.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/678)*

**Comment by randomnote1** [comment]
I tend to agree with you, however when I try to load the class after Pester completes its tests, the class isn't available. Yes, I agree with you there. I'll get moving on the workaround.

---

Review status: 0 of 9 files reviewed at latest revision, 5 unresolved discussions.

---

*[DSCClassResources/MSFT_xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership/xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership.psd1, line 1 at r3](https://reviewable.io:443/reviews/powershell/xsqlserver/678#-KokI97jN4jfa7bmus0s:-KolZPJeAXK6RcxyUePl:bq2c41c) ([raw file](https://github.com/powershell/xsqlserver/blob/1c5bdba81a7d2485bfa2b3f545dacdc6257c388e/DSCClassResources/MSFT_xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership/xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership.psd1#L1)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Note that I changed to .psm1 for NestedModules instead of .psd1 when I removed this file.
</blockquote></details>

Yup, I did the same thing.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/678#-:-KolUJDiXQDOZD50e1Zs:bg5g7o7)*

**Comment by johlju** [comment]
@randomnote1 Great that you are working on the workaround!

I trying to create a mockup of this problem with pester so we can submit an issue (somewhere) with an example containing so little code as possible. 😄

**Comment by johlju** [comment]
@randomnote1 I fixed a mockup here https://github.com/johlju/PesterTestClassBasedResource that uses minimal code. I can submit an issue to Pester repository.

**Comment by johlju** [comment]
@randomnote1 There seem there is a bug fix in PowerShell 6.0. I can run the mockup on macOS in PowerShell 6.0 beta 3. I will deploy PowerShell 6.0 beta 3 on a Windows Server to see if that works.

Issue in Pester: https://github.com/pester/Pester/issues/797

**Comment by johlju** [comment]
@randomnote1 Update the Pester issue pester/Pester#797.

**Comment by randomnote1** [comment]
Glad to see there is a fix in progress! Not gonna help us with PowerShell v5 though.

**Comment by johlju** [comment]
Nope, unfortunately not :)

**Comment by johlju** [comment]
Katie merged the fix for DscResource.Tests so that it can find resources under DscClassResources. So you can change back to the original AppVeyor.yml file now.

**Comment by randomnote1** [comment]
I created a new pull request for the MOF based version of this in PR #682. We can keep this one open to tinker with the class based resource if you would like.

**Comment by johlju** [comment]
Lets think of what we do with this. I do want to keep the class-based resource to test the maturity of class-based resources vs Pester tests down the line. 
Thought about if I should branch the dev Branch here in this repo or if I should keep it in a fork.

**Comment by johlju** [comment]
I branched dev to test-problem-class-based-resource-issue-#673 and moved this PR to point to that as base branch instead.

labeled this as 'on hold' for the time being.

File: appveyor.yml [hunk 1]
```diff
@@ -3,7 +3,11 @@
 #---------------------------------#
 version: 6.0.{build}.0
 install:
-    - git clone https://github.com/PowerShell/DscResource.Tests
+    - git clone https://github.com/johlju/DscResource.Tests
```
Comment (johlju) [inline appveyor.yml]: Could you revert lin 6-10 back to 
```
- git clone https://github.com/PowerShell/DscResource.Tests
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,34 @@
 - Added new resource
   - xSQLServerAlwaysOnAvailabilityGroupDatabaseMembership
   - Added localization support for all strings.
+  - Moved the resource to the DSCClassResources directory.
+  - Moved the Import-SQLPSModule command from the constructor to the Set method.
+- Changes to xSQLServerRSConfig
```
Comment (johlju) [inline CHANGELOG.md]: Could you rebase this one against the new base branch `test-problem-class-based-resource-issue-#673`.

**Comment by johlju** [comment]
@randomnote1 If you check 'Allow edits from maintainers' I can fix this PR myself. So you don't have to. You deserve som DSC rest now! You did awesome work these days. 😄

**Comment by randomnote1** [comment]
I thought I had that checked already. All yours!

**Comment by johlju** [comment]
:lgtm:

This one will be merged into the base-branch `test-problem-class-based-resource-issue-#673` to have a class-based resource to test running together witht he common-test, and verify that it will work to run the tests more than once in a row (issue #673). This will probably not happen until PowerShell 6.0 is released.

---

Reviewed 5 of 17 files at r3, 9 of 10 files at r4, 1 of 1 files at r5, 2 of 2 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xsqlserver/678#-:-KovYscJQ7PHqpP7FnxP:bbecapb)*

**Comment by johlju** [comment]
@randomnote1 This is merged. Now you can delete you working branch whenever you feel like it.

**Comment by randomnote1** [comment]
Looks like PowerShell 6 won't be released.

Microsoft Transitioning Windows PowerShell 6.0 into PowerShell Core, shared via News Pro, https://redmondmag.com/articles/2017/07/17/windows-powershell-6.0-and-powershell-core.aspx?admgarea=BDNA
Install here: https://newspro.microsoft.com

**Pull request merged**